src/python/WMComponent/WorkQueueManager/WorkQueueManagerCleaner.py

```diff
@@ -47,9 +53,14 @@ def algorithm(self, parameters):
             # this will clean up whatever left over from above clean up.
             # also if the wq replication has problem it won't delay the killing jobs in condor
             # and updating wmbs status
-            abortedAndForceCompleteRequests = self.abortedAndForceCompleteWorkflowCache.getData()
+            # state lists which shouldn't be populated in wmbs. (To prevent creating work before WQE status updated)
+            # added completed list due to the race condition (The extra cost of the query is ~1sec per request)
```

Comment:
I guess this comment (1sec per request) is no longer true, is it? We only fetch request names given a list of request status (thus a single reqmgr2 http call).

Reply:
Yes I will remove that. Thanks